### PR TITLE
Calliope mini

### PR DESCRIPTION
**Pull request by jhmaloney**
Add a board variant for the Calliope mini, the German variation of the BBC micro:bit.

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -386,6 +386,51 @@ BBCmicrobit.menu.softdevice.s130.upload.maximum_size=151552
 BBCmicrobit.menu.softdevice.s130.build.extra_flags=-DNRF51 -DS130 -DNRF51_S130
 BBCmicrobit.menu.softdevice.s130.build.ldscript=armgcc_s130_nrf51822_xxaa.ld
 
+Calliope.name=Calliope mini
```
Comment (sandeepmistry) [inline boards.txt]: What do you think about changing the key to `CalliopeMini`. If they are successful could have more than one board?

**Comment by sandeepmistry** [comment]
Closes #204.

**Comment by sandeepmistry** [comment]
Two other minors things:
* please add a link to the read me for the board
* also update the `.travis.yml` for the new variant.

**Comment by jhmaloney** [comment]
I'll change the name to "CalliopeMini" (with underscores where appropriate). Will also add an entry to the .travis.yml file.

Re: please add a link to the read me for the board

Happy to, but not sure what's needed. By "read me" do you mean a link to the Calliope website? Or should I create a README file and put it somewhere? If so, where?

**Comment by jhmaloney** [comment]
I think I figured out what was need re: README and I've made all the requested changes.

File: .travis.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ script:
   - buildExampleSketch sandeepmistry:nRF5:BLENano:version=1_0 01.Basics Blink
   - buildExampleSketch sandeepmistry:nRF5:RedBearLab_nRF51822:version=1_0 01.Basics Blink
   - buildExampleSketch sandeepmistry:nRF5:BBCmicrobit 01.Basics Blink
+  - buildExampleSketch sandeepmistry:nRF5:Calliope_Mini 01.Basics Blink
```
Comment (sandeepmistry) [inline .travis.yml]: This should match key in boards.txt: CalliopeMini

File: variants/CalliopeMini/variant.h [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#ifndef _VARIANT_CALLIOPE_
+#define _VARIANT_CALLIOPE_
```
Comment (sandeepmistry) [inline variants/CalliopeMini/variant.h]: `_VARIANT_CALLIOPE_MINI_` here?

**Comment by sandeepmistry** [comment]
@jhmaloney great!

@dlabun if this looks good to you, we can "squash and merge" using Github.

**Pull request merged**